src/Remapping/interpolate_array.jl

```diff
@@ -590,30 +590,33 @@ function interpolate_column(
     # we have to make sure that we are passing around views of the same array (as opposed to
     # copied of it).
     if physical_z
-        # We are hardcoding the transformation from Hypsography.LinearAdaption
-        space.hypsography isa Hypsography.LinearAdaption ||
-            error("Cannot interpolate $(space.hypsography) hypsography")
-
         FT = Spaces.undertype(axes(field))
 
         # interpolate_slab! takes a vector
         z_surface = [zero(FT)]
 
         interpolate_slab!(
             z_surface,
-            space.hypsography.surface,
+            space.hypsography.surface.z,
             [Fields.SlabIndex(nothing, gidx)],
             [Is],
         )
         z_surface = z_surface[]
         z_top = Spaces.vertical_topology(space).mesh.domain.coord_max.z
 
-        zpts_ref = [
-            Geometry.ZPoint((z.z - z_surface) / (1 - z_surface / z_top)) for
-            z in zpts if z.z > z_surface
+        all_zpts_ref = [
+            Hypsography.physical_z_to_ref_z(
+                Spaces.grid(space).hypsography,
+                z_ref,
+                Geometry.ZPoint(z_surface),
+                Geometry.ZPoint(z_top),
+            ) for z_ref in zpts
```

Comment:
Should this be `map(z_ref ->Hypsography.physical_z_to_ref_z(
                Spaces.grid(space).hypsography,
                z_ref,
                Geometry.ZPoint(z_surface),
                Geometry.ZPoint(z_top),
            ), zpts)`
instead?

Reply:
They are both ugly in my opinion, but if you prefer the map, I'll switch to map.

Reply:
I think `map` preserves the input type, whereas `[generator]` will change the type, so it's preferable in that sense.

Reply:
e.g.,:
```julia
julia> map(ntuple(i->i, 4)) do j
       j+1
       end
(2, 3, 4, 5)

julia> [j+1 for j in ntuple(i->i, 4)]
4-element Vector{Int64}:
 2
 3
 4
 5
```

Reply:
Got it, thanks!

Reply:
I am still used to the Pythonic way

Reply:
I also changed the comprehension below to a `filter!`